src/tests/integration/test-data/src/Unit.t.sol

```diff
@@ -242,6 +242,59 @@ contract UnitTest is Test {
         assertApproxEqAbs(int_zero, neg_a, maxDelta, err);
     }
 
+    function test_assertApproxEqRel_uint_unit() public pure {
+        uint256 zero = 0;
+        uint256 a = 8;
+        uint256 b = 10;
+        uint256 percentDelta = 1e18;
+        string memory err = "throw test";
```

Comment:
Maybe it'd be useful to add tests checking that the error `err` is returned correctly, e.g., similar to https://github.com/runtimeverification/kontrol/blob/a551f742db727cd23c989c4c761b37a0e0821a7f/src/tests/integration/test-data/src/Unit.t.sol#L60-L63

Reply:
Yes, I agree. Will remove them from the tests and put them in separate tests.

Reply:
Hi @palinatolmach. I have updated the tests in https://github.com/runtimeverification/kontrol/pull/593/commits/01dda9d25ab4b0a9247e0e7c18d895bd9651d099. If the changes are okay, please feel free to resolve this particular issue.

Reply:
Thanks @jinxinglim! Do some of these new tests pass as expected, e.g., `test_assertApproxEqAbs_uint_err`? Right now, they'll all be skipped as they're all placed in the `...-skip` file.

Reply:
I ran `forge test` on them locally and they passed. But have not tried them with `kontrol prove`. There are `test_*_err` tests that are placed in both files. So I am unsure what's the criteria to place tests in those files. Are the tests in `*-all` supposed to pass `forge test` or `kontrol prove`?

Reply:
All tests in `*-all` are supposed to pass with `kontrol prove`, these are tests that are executed with `kontrol prove` on CI to make sure all new changes are implemented correctly and don't break any of the existing functionality. For features that are available in both Foundry and Kontrol, they should pass with both commands :). But, generally, all tests are concerned with Kontrol. 

You can add tests to `*-skip` if they tests functionality that, e.g., has not been implemented yet, so we can't include those in CI runs. But all tests that exercise new functionality that works are to only be placed in `*-all`, this way they'll be run by CI on both booster and legacy backends. Ideally, they should take not more than several minutes to run, and, if they're time consuming, please add them to `*-skip-legacy` so they're only run on the booster backend, and the CI run time doesn't increase significantly. There's also come explanation here: https://github.com/runtimeverification/kontrol/blob/master/CONTRIBUTING.md. Hope that clarifies!

Reply:
Okay. Got it. Thank you! Let me test the new `test_*_err` tests with `kontrol prove` and place them accordingly.

Reply:
Hi @palinatolmach. I just ran a few of the new `test_*_err` tests and all of them returned the same error as the one seen in issue https://github.com/runtimeverification/kontrol/issues/581, which is yet to resolve. The new `test_*_err` tests only passed at `forge test` but not `kontrol prove`. So I will place them in `end-to-end-skip`. Is this okay?

Reply:
I don't think you've missed anything! Looks like it's an SMT error caused by CI flakiness, I triggered a re-run, so let's see. I see about the https://github.com/runtimeverification/kontrol/issues/581 failure, thanks! I agree with adding tests to `end-to-end-skip`, but it'd be good to figure out why that happens, let's check with the backend tomorrow.